nomad/state/state_store.go

```diff
@@ -3582,8 +3606,13 @@ func (s *StateStore) UpdateAllocsFromClient(msgType structs.MessageType, index u
 	txn := s.db.WriteTxnMsgT(msgType, index)
 	defer txn.Abort()
 
+	// Capture all nodes being affected. Alloc updates from clients are batched
+	// so this request may include allocs from several nodes.
+	nodeIDs := make(map[string]interface{})
+
 	// Handle each of the updated allocations
 	for _, alloc := range allocs {
+		nodeIDs[alloc.NodeID] = struct{}{}
```

Comment:
Seems like this is being used only as a set to track NodeID? If so consider using a `set.Set`